src/host/posix/dnssd.hpp

```diff
@@ -391,10 +367,16 @@ class DnssdPlatform : public Mdns::StateObserver, private NonCopyable
 
         IteratorType FindEntry(uint64_t aInfraIfIndex, const CallbackType &aCallback)
         {
-            return std::find_if(mEntries.begin(), mEntries.end(),
-                                [aInfraIfIndex, &aCallback](const RequestType &entry) {
-                                    return entry.first == aInfraIfIndex && *entry.second == aCallback;
-                                });
+            IteratorType iter;
+
+            for (iter = mEntries.begin(); iter != mEntries.end(); iter++)
+            {
+                if (iter->first == aInfraIfIndex && *(iter->second) == aCallback)
+                {
+                    break;
+                }
+            }
+            return iter;
```

Comment:
Just curious, why didn't this continue using `std::find_if`?

Reply:
I found the original code is less readable.